README.md

```diff
@@ -12,46 +12,57 @@ oc process -f https://raw.githubusercontent.com/jaegertracing/jaeger-openshift/m
 Once everything is ready, `oc status` tells you where to find Jaeger URL.
 
 ## Production setup
-This template deploys all Jaeger components as a separate services: StatefulSet Cassandra storage, agent as DaemonSet,
-collector and query service with UI. Each one of those can be managed and scaled individually. Because this template
-deploys the agent as a DaemonSet it requires system:admin permissions, therefore cannot be used in OpenShift online.
+This template deploys the Collector, Query Service (with UI) and Cassandra storage (StatefulSet) as separate individually scalable services.
 
-Install everything in `jaeger` namespace:
+Install everything in `jaeger-infra` namespace:
 ```bash
-oc login -u system:admin
-oc new-project jaeger
-oc process -f https://raw.githubusercontent.com/jaegertracing/jaeger-openshift/master/production/jaeger-production-template.yml | oc create -n jaeger -f -
-```
-
-Give user `developer` permissions to access `jaeger` namespace:
-```bash
-oc policy add-role-to-user cluster-admin developer -n jaeger
+oc new-project jaeger-infra
+oc process -f https://raw.githubusercontent.com/jaegertracing/jaeger-openshift/master/production/jaeger-production-template.yml | oc create -n jaeger-infra -f -
 ```
 
 Note that it's OK to have the Query and Collector pods to be in an error state for the first minute or so. This is
 because these components attempt to connect to Cassandra right away and hard fail if they can't after N attempts.
 
-Your Agent hostname is `jaeger-agent.${NAMESPACE}.svc.cluster.local`
-
-### Install Jaeger as `developer` user
-
-Jaeger can also be installed as `developer` (or any other user). As the DaemonSet runs at the node level,
-however, your user need to have such permission. It can be achieved by running this:
-
-```bash
-oc login -u developer
-oc new-project jaeger
-
-oc login -u system:admin
-oc create -f https://raw.githubusercontent.com/jaegertracing/jaeger-openshift/master/production/daemonset-admin.yml
-oc adm policy add-role-to-user daemonset-admin developer -n jaeger // jaeger namespace has been already created and it is accessible by developer user
+### Deploying the agent as sidecar
+
+The Jaeger Agent is designed to be deployed local to your service, so that it can receive traces via UDP keeping your
+application's load minimal. As such, it's ideal to have the Agent to be deployed as a sidecar to your application's pod.
+For instance, assuming that your application is named `myapp` and the image is for it is `openshift/hello-openshift`, your
+`Deployment` descriptor would be something like:
+
+```yaml
+- apiVersion: extensions/v1beta1
+  kind: Deployment
```

Comment:
that section is about best practices :)
 
Generally speaking, some might use `StatefulSet` with agent (e.g. I have instrumented cassandra with jaeger). My point is that this can be added to every entity supporting `spec.containers` (I guess).